### PR TITLE
[RLlib] Trainer runner quick cleanups

### PR DESCRIPTION
**Pull request by kouroshHakha**
## Why are these changes needed?
Some quick clean ups before we dive deeper into Trainer Runner's interface. 

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [x] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: rllib/core/rl_trainer/tests/test_trainer_runner.py [hunk 1]
```diff
@@ -0,0 +1,153 @@
+import gymnasium as gym
```
Comment (kouroshHakha) [inline rllib/core/rl_trainer/tests/test_trainer_runner.py:1]: This is for tf only. Like the other RLTrainer unittests we will integrate torch tests into the same file, once we implement the class.

**Review by avnishn — APPROVED** [review]
LGTM

File: rllib/BUILD [hunk 2]
```diff
@@ -1818,14 +1818,21 @@ py_test(
     srcs = ["core/rl_module/tests/test_marl_module.py"]
 )
 
-
+# to be removed
```
Comment (gjoliver) [inline rllib/BUILD:1821]: what is this? if it's a TODO, please add TODO and your handle.

**Review by gjoliver — APPROVED** [review]

File: rllib/BUILD (hunk 2 above)
Comment (kouroshHakha) [reply rllib/BUILD]: yes sir. Fixed.

**Comment by kouroshHakha** [comment]
@gjoliver Let's see if they pass this time.

**Comment by kouroshHakha** [comment]
@gjoliver should be ready for merge.

File: rllib/core/rl_trainer/trainer_runner.py [hunk 3]
```diff
@@ -101,28 +108,22 @@ def _compute_necessary_resources(self):
...
-                batch_to_send = {}
-                for pid, sub_batch in batch.policy_batches.items():
-                    batch_size = math.ceil(len(sub_batch) / global_size)
-                    start = batch_size * i
-                    end = min(start + batch_size, len(sub_batch))
-                    batch_to_send[pid] = sub_batch[int(start) : int(end)]
-                new_batch = MultiAgentBatch(batch_to_send, int(batch_size))
-                refs.append(worker.update.remote(new_batch))
+        global_size = len(self.workers)
+        batch_size = math.ceil(len(batch) / global_size)
```
Comment (gjoliver) [inline rllib/core/rl_trainer/trainer_runner.py:117]: why do we need to calculate this size here, unused, then immediately get updated below?

**Review by gjoliver — APPROVED** [review]

**Comment by gjoliver** [comment]
looked at the failed tests, mem leak ones and r2d2 are not relevant. recsim test also timed out. but there is very little chance it has anything to do with this PR.
merging.

**Pull request merged**